app/models/iiif_presentation.rb

```diff
@@ -1,39 +1,45 @@
 # frozen_string_literal: true
 
-class IiifManifestFactory
-  attr_reader :oid, :manifest, :parent_object, :mets_doc
+class IiifPresentation
+  attr_reader :parent_object, :mets_doc, :oid
 
   require 'iiif/presentation'
 
   def initialize(oid)
     @manifest_base_url = ENV["IIIF_MANIFESTS_BASE_URL"] || "http://localhost/manifests/"
     @image_base_url = ENV["IIIF_IMAGE_BASE_URL"] || "http://localhost:8182/iiif"
-    @oid = oid
     @parent_object = ParentObject.find_by(oid: oid)
-    @manifest = construct_manifest
+    @oid = oid
+  end
+
+  def validate_images
+    raise "There are no child objects for #{@oid}" if ChildObject.where(parent_object: parent_object).order(:order).empty?
```

Comment:
This would be a better approach:

```suggestion
    if ChildObject.where(parent_object: parent_object).order(:order).empty?
      errors.add(:manifest, 'There are no child objects for #{@oid}') 
      return false
    end
    true
```

Then you don't have to recover from a raise